cmd/bosun/sched/check.go

```diff
@@ -111,7 +111,20 @@ func (s *Schedule) RunHistory(r *RunHistory) {
 		if event.Unevaluated {
 			continue
 		}
+		prev := state.Last()
 		state.Append(event)
+		if prev.IncidentId != 0 {
+			//if last event has incident id and is not closed, we continue it.
+			s.incidentLock.Lock()
+			if incident, ok := s.Incidents[prev.IncidentId]; ok && incident.End.IsZero() {
+				event.IncidentId = prev.IncidentId
+			}
+			s.incidentLock.Unlock()
+		} else if event.Status != StNormal {
+			//otherwise create new incident on first non-normal event
```

Comment:
comments have a space after `//`, start with caps, and end with `.`: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences